TestLib/Exception/GREYDefaultFailureHandler.m

```diff
@@ -81,24 +93,57 @@ - (void)handleException:(GREYFrameworkException *)exception details:(NSString *)
       [GREYFailureScreenshotSaver saveFailureScreenshotsInDictionary:appScreenshots
                                                          toDirectory:screenshotDir];
   NSAssert(screenshotPaths, @"Screenshots must be present");
-  NSArray *stackTrace = [NSThread callStackSymbols];
+  return screenshotPaths;
+}
 
-  NSString *appUIHierarchy = [exception.userInfo valueForKey:kErrorDetailAppUIHierarchyKey];
-  // For calls from GREYAsserts in the test side, the hierarchy must be populated here.
-  if (!appUIHierarchy) {
-    appUIHierarchy = [GREYElementHierarchy hierarchyString];
+- (NSString *)consoleLogForException:(GREYFrameworkException *)exception
+                             details:(NSString *)details
+                     screenshotPaths:(GREYFailureScreenshots *)screenshotPaths
+                      appUIhierarchy:(NSString *)appUIHierarchy
+                     currentTestCase:(id)currentTestCase
+                          stackTrace:(NSArray *)stackTrace {
+  if ([exception.reason containsString:@"the desired element was not found"]) {
+    /// As GREYErrorFormatter adds support for more error codes,
+    /// this check will not be needed, and this formatted output will be used for every log
+    NSMutableString *formattedOutput = [details mutableCopy];
+    /// Appends screenshots and ui hierarchy to the console output if they didn't already exist.
+    if (!exception.userInfo[kErrorDetailAppScreenshotsKey]) {
+      for (NSString *key in screenshotPaths.allKeys) {
+        [formattedOutput appendString:[NSString stringWithFormat:@"\n%@: %@\n",
+                                       key, screenshotPaths[key]]];
+      }
+    }
+    if (!exception.userInfo[kErrorDetailAppUIHierarchyKey]) {
+    [formattedOutput appendString:[NSString stringWithFormat:@"\n%@\n",
```

Comment:
Use [`-[NSMutableString appendFormat:]`](https://developer.apple.com/documentation/foundation/nsmutablestring/1497308-appendformat?language=objc). please apply this to all other changes.